src/dmtcpplugin.cpp

```diff
@@ -169,10 +169,8 @@ dmtcp_get_tmpdir(void)
 EXTERNC const char *
 dmtcp_get_ckpt_dir()
 {
-  static string tmpdir;
-
-  tmpdir = ProcessInfo::instance().getCkptDir();
-  return tmpdir.c_str();
+  static string *tmpdir = new string(ProcessInfo::instance().getCkptDir());
```

Comment:
@gc00: I created a new PR (https://github.com/dmtcp/dmtcp/pull/936) with this commit, along with a few more changes from static string to heap string. The commit message has some more information now. 

For background, Rohan had removed most of the static strings a long time ago because of the deadlock caused by their constructor/at-exit handler. The ones from their PR were overlooked at the time or added later.

Reply:
In the new PR #936, can you add a comment about why a static string would create a deadlock?  It's not obvious to me.  Without a comment, this code will be difficult to maintain.

Reply:
Added a comment.